files/class-api-client.php

```diff
@@ -21,12 +23,19 @@ class API_Client {
 	private $files_site_id;
 	private $files_token;
 
-	public function __construct( $api_base, $files_site_id, $files_token ) {
+	/**
+	 * @var API_Cache
+	 */
+	private $cache;
+
+	public function __construct( $api_base, $files_site_id, $files_token, $cache ) {
```

Comment:
How do we suggest handling direct instantiation of this class? i.e. `new API_Client( ... )` Should the caller just pass in `API_Cache::get_instance()`. Is it worth creating an interface or abstract class and typecasting the param?

Reply:
I guess it depends on the likelihood of there being other Client types or Cache types. On the other hand, using interfaces might make writing tests for these much simpler

Reply:
No, I wouldn't worry too much about it.